app/helpers/react_components/community/video_grid.rb

```diff
@@ -16,7 +16,8 @@ def to_s
             initial_data:
           }
         },
-        tracks:
+        tracks:,
+        selected_track_slug: params[:track_slug]
```

Comment:
There was no logic for selected track. I added this (and the bit below) but this might be a terrible way to do it, so feel free to rip it apart and redo it.